cli/src/commands/run_config_docker.py

```diff
@@ -12,35 +13,70 @@ def get_usage(self):
+        return [{"name": "config",
+                 "description": "path to the config to run from the remote docker image"},
+                {"name": "from_local_code",
+                 "description": "use the local code to run the crawl",
+                 'optional': False}]
+
+    @staticmethod
+    def from_local_code_parse(args):
+        print(args)
+        if len(args) < 2:
+            return False
+        if isinstance(args[1], bool):
+            return args[1]
+        return isinstance(args[1], str) and args[1].lower() == 'true'
 
     def run(self, args):
-        from os import getcwd, environ
 
         self.check_not_docsearch_app_id('run a config manually')
 
-        self.exec_shell_command(["docker", "stop", "documentation-scrapper-dev"])
-        self.exec_shell_command(["docker", "rm", "documentation-scrapper-dev"])
-
-        f = open(args[0], 'r')
-        config = f.read()
-
-        run_command = [
-            'docker',
-            'run',
-            '-e',
-            'APPLICATION_ID=' + environ.get('APPLICATION_ID'),
-            '-e',
-            'API_KEY=' + environ.get('API_KEY'),
-            '-e',
-            "CONFIG=" + config,
-            '-v',
-            getcwd() + '/scraper/src:/root/src',
-            '--name',
-            'documentation-scrapper-dev',
-            '-t',
-            'algolia/documentation-scrapper-dev',
-            '/root/run'
-        ]
+        if os.path.isfile(args[0]):
+            f = open(args[0], 'r')
+            config = f.read()
+        else:
+            raise ValueError(
+                "Config option: {} is not a path to a file".format(args[0]))
+
+        from_local_code = self.get_option('from_local_code', args)
 
+        run_command = []
+        if not from_local_code:
+            run_command = [
+                'docker',
+                'run',
+                '--rm',
+                '-e',
+                'APPLICATION_ID=' + os.environ.get('APPLICATION_ID'),
+                '-e',
+                'API_KEY=' + os.environ.get('API_KEY'),
+                '-e',
+                "CONFIG=" + config,
+                '--name',
+                'documentation-scraper',
+                '-t',
+                'algolia/documentation-scraper',
+                '-i',
+                '/root/run'
+            ]
+        else:
+            run_command = [
+                'docker',
+                'run',
+                '--rm',
+                '-e',
+                'APPLICATION_ID=' + os.environ.get('APPLICATION_ID'),
+                '-e',
+                'API_KEY=' + os.environ.get('API_KEY'),
+                '-e',
+                "CONFIG=" + config,
+                '-v',
+                os.getcwd() + '/scraper/src:/root/src',
+                '--name',
+                'documentation-scraper-dev',
+                '-t',
+                'algolia/documentation-scraper-dev',
+                '/root/run'
+            ]
```

Comment:
Both calls above are pretty similar. The only different is that the dev one mounts  a volume and uses a different image.

What do you think about putting the common commands in one variable, and then only adding the additional commands.
Something like:

```pythonapplication_id = os.environ.get("APPLICATION_ID")
api_key = os.environ.get("API_KEY")

if from_local_code:
    container_name = "documentation-scraper-dev"
    image_name = "algolia/documentation-scraper-dev"
else:
    container_name = "documentation-scraper"
    image_name = "algolia/documentation-scraper"

run_command = [
    "docker",
    "run",
    "--rm",
    "-e",
    "APPLICATION_ID=" + application_id,
    "-e",
    "API_KEY=" + api_key,
    "-e",
    "CONFIG=" + config,
    "--name",
    container_name,
    "-t",
    image_name,
]

if from_local_code:
    run_command.append("-v")
    run_command.append(os.getcwd() + "/scraper/src:/root/src")
else:
    run_command.append("-i")

run_command.append("/root/run")
```

This could make it easier to spot the difference between dev and prod

Reply:
Great idea, I will implement it.